### PR TITLE
Fix #622: rework https://github.com/MiguelCastillo/Brackets-InteractiveLinter to load in browser

### LINKED ISSUE #622: Rework Brackets-InteractiveLinter extension for Bramble
From https://github.com/mozilla/thimble.mozilla.org/issues/1238#issuecomment-284078401, it would be cool to try and get https://github.com/MiguelCastillo/Brackets-InteractiveLinter working in Bramble.

The main problem is going to be swapping out the *load files from the filesystem* logic to instead use `require()` calls, and statically bundle all the things you need.

Whoever wants to try this, I can walk you through what needs to happen.  It's not trivial, but it's not that hard.

### PR DESCRIPTION
**Pull request by humphd**
This is partially working now:

screen shot 2017-03-04 at 5 11 55 pm

I need to figure out how to get the eslint, csslint, and jsonlint to actually show warnings.  I suspect it's trying to read non-existent `rc` files for them.  I'll do some more work on it this week.

### CODE REVIEW

**Comment by humphd** [comment]
OK, it works now for JS, CSS, HTML, and JSON files.  @flukeout you should pull this branch when you have time and test it locally, to see what you think.  I don't love the light bulb icon, or the styling in general, but that could be improved.  It's sort of an advanced thing, especially with the level of error messages.  However, it's very cool!

@Pomax it would probably be pretty easy to rework the SlowParse stuff to "Just Work" with this extension, since all the linters run in workers as separate plugins already.

**Comment by Pomax** [comment]
I don't think Slowparse can work in a worker, as it relies on `document.createDocumentFragment`

File: src/extensions/default/Brackets-InteractiveLinter/.brackets.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "interactive-linter.delay": 500,
```
Comment (humphd) [inline src/extensions/default/Brackets-InteractiveLinter/.brackets.json]: We need to add `"interactive-linter.enabled": false,` here, then use that

File: src/extensions/default/Brackets-InteractiveLinter/linterManager.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+ */
+
+
+define(function (require /*, exports, module*/) {
+    "use strict";
+
+    var _                  = brackets.getModule("thirdparty/lodash"),
+        LanguageManager    = brackets.getModule("language/LanguageManager"),
+        PreferencesManager = brackets.getModule("preferences/PreferencesManager"),
+        preferences        = PreferencesManager.getExtensionPrefs("interactive-linter"),
```
Comment (humphd) [inline src/extensions/default/Brackets-InteractiveLinter/linterManager.js]: We might need to fix some code here to have it `require` vs. loading `brackets.json`, which I think this does.

**Comment by humphd** [comment]
One of my students is going to work on this.  We'll land it default to off, and make it possible in Thimble to enable it.

**Comment by flukeout** [comment]
@flukeout to try out and determine if we want to keep this open or not.

**Comment by flukeout** [comment]
@humphd will rebase.

**Comment by humphd** [comment]
@flukeout no rush on this, but I've rebased it if you want to try it out locally.  It's working for me.

**Comment by Pomax** [comment]
do we still want to try to land this PR?

**Comment by humphd** [comment]
No, let's close it.  We can do the slowparse stuff if we do anything.

**Pull request closed without merging**